src/commands/deploy/metadata/validate.ts

```diff
@@ -0,0 +1,143 @@
+      summary: messages.getMessage('flags.source-dir.summary'),
+      multiple: true,
+      exclusive: ['manifest', 'metadata'],
+      exactlyOne: ['manifest', 'source-dir', 'metadata'],
+    }),
+    'target-org': Flags.requiredOrg({
+      char: 'o',
+      description: messages.getMessage('flags.target-org.description'),
+      summary: messages.getMessage('flags.target-org.summary'),
+    }),
+    tests: Flags.string({
+      char: 't',
+      multiple: true,
+      summary: messages.getMessage('flags.tests.summary'),
+    }),
+    'test-level': testLevelFlag({
+      options: [TestLevel.RunAllTestsInOrg, TestLevel.RunLocalTests, TestLevel.RunSpecifiedTests],
+      default: TestLevel.RunLocalTests,
+      description: messages.getMessage('flags.test-level.description'),
+      summary: messages.getMessage('flags.test-level.summary'),
+    }),
+    verbose: Flags.boolean({
+      summary: messages.getMessage('flags.verbose.summary'),
+    }),
+    wait: Flags.duration({
+      char: 'w',
+      summary: messages.getMessage('flags.wait.summary'),
+      description: messages.getMessage('flags.wait.description'),
+      unit: 'minutes',
+      defaultValue: 33,
+      helpValue: '<minutes>',
+      min: 1,
+    }),
+  };
+
+  public static configurationVariablesSection = toHelpSection(
+    'CONFIGURATION VARIABLES',
+    OrgConfigProperties.TARGET_ORG,
+    SfdxPropertyKeys.API_VERSION
+  );
+
+  public static envVariablesSection = toHelpSection(
+    'ENVIRONMENT VARIABLES',
+    EnvironmentVariable.SF_TARGET_ORG,
+    EnvironmentVariable.SF_USE_PROGRESS_BAR
+  );
+
+  public async run(): Promise<DeployMetadataValidateResult> {
+    const flags = (await this.parse(DeployMetadataValidate)).flags;
+    const api = resolveRestDeploy();
+    const { deploy, componentSet } = await executeDeploy({
+      ...flags,
+      'dry-run': true,
+      'target-org': flags['target-org'].getUsername(),
+      api,
+    });
+
+    this.log(getVersionMessage('Validating Deployment', componentSet, api));
+    this.log(`Deploy ID: ${deploy.id}`);
+    new DeployProgress(deploy, this.jsonEnabled()).start();
+
+    const result = await deploy.pollStatus(500, flags.wait.seconds);
+    this.setExitCode(result);
+
+    if (!this.jsonEnabled()) {
+      displayDeployResults(result, flags['test-level'], flags.verbose);
+    }
+
+    if (result.response.status === RequestStatus.Succeeded) {
+      this.log();
+      this.log(messages.getMessage('info.SuccessfulValidation', [deploy.id]));
+
+      const suggestedCommand = `${this.config.bin} deploy metadata quick --job-id ${deploy.id}`;
+      this.log(`\nRun ${bold(suggestedCommand)} to execute this deploy.`);
+    } else {
+      throw messages.createError('error.FailedValidation', [deploy.id]);
+    }
+
+    return {
+      jobId: result.response.id,
+      files: result.getFileResponses() || [],
+      tests: getTestResults(result.response),
+    };
+  }
+
+  private setExitCode(result: DeployResult): void {
```

Comment:
can we keep the current pattern, where there's 0 and 1 but also partialSuccess and timeout exit codes?
https://salesforce.quip.com/JoC2A2IlF9a3

then a CI could make a smarter decision based on the exit code (esp for handling timeout)